numpy/core/tests/test_multiarray.py

```diff
@@ -6726,6 +6727,27 @@ def test_null_inside_ustring_array_is_truthy(self):
         a[0] = ' \0 \0'
         self.assertTrue(a)
 
+
+class TestCTypes(TestCase):
+
+    def test_ctypes_is_available(self):
+        test_arr = np.array([[1, 2, 3], [4, 5, 6]])
+
+        self.assertEqual(ctypes, test_arr.ctypes._ctypes)
+        assert_equal(test_arr.shape, (2, 3))
+
+    def test_ctypes_is_not_available(self):
+        _internal.ctypes = None
+        try:
+            test_arr = np.array([[1, 2, 3], [4, 5, 6]])
+
+            self.assertIsInstance(
+                test_arr.ctypes._ctypes, _internal._missing_ctypes)
```

Comment:
I'm almost tempted to spin off a new python interpreter without ctypes to run the test, to ensure it can't break anything else

You might also be able to get your `sys.modules` hackery to work if you try to put everything back together again, but also a little risky.